From ff5535940dd9038fdbcc22139dd8fd5cc56423e3 Mon Sep 17 00:00:00 2001
From: Lars Greiss <kufikugel@googlemail.com>
Date: Mon, 1 Dec 2014 15:14:15 +0100
Subject: [PATCH 08/17] Dialer: We want consistents....change CheckBoxes to
 Switches

Like in settings it is on other places often switches and here and there a checkbox
Change all CheckBoxes to Switches like sugested by google for new LP Ui.

Thanks for our users which point us to this small visual changes.

Change-Id: Iaf11e591f377d2f6be07bad052338ed38d6af119
Signed-off-by: Chet Kener <Cl3Kener@gmail.com>
---
 res/xml/general_settings.xml                           | 10 +++++-----
 .../dialer/settings/GeneralSettingsFragment.java       | 18 +++++++++---------
 2 files changed, 14 insertions(+), 14 deletions(-)

diff --git a/res/xml/general_settings.xml b/res/xml/general_settings.xml
index 1751e17..406927c 100644
--- a/res/xml/general_settings.xml
+++ b/res/xml/general_settings.xml
@@ -47,13 +47,13 @@
             android:persistent="false"
             android:ringtoneType="ringtone" />
 
-        <CheckBoxPreference
+        <SwitchPreference
             android:key="button_play_dtmf_tone"
             android:title="@string/dtmf_tone_enable_title"
             android:persistent="false"
             android:defaultValue="true" />
 
-        <CheckBoxPreference
+        <SwitchPreference
             android:key="button_vibrate_on_ring"
             android:title="@string/vibrate_on_ring_title"
             android:persistent="false"
@@ -65,15 +65,15 @@
         android:key="dialer_general_incall_vibration_category_key"
         android:title="@string/incall_vibration_category_title">
 
-        <CheckBoxPreference
+        <SwitchPreference
             android:key="incall_vibrate_outgoing"
             android:title="@string/incall_vibrate_outgoing_title" />
 
-        <CheckBoxPreference
+        <SwitchPreference
             android:key="incall_vibrate_hangup"
             android:title="@string/incall_vibrate_hangup_title" />
 
-        <CheckBoxPreference
+        <SwitchPreference
             android:key="incall_vibrate_45"
             android:title="@string/incall_vibrate_45_title"
             android:summary="@string/incall_vibrate_45_summary" />
diff --git a/src/com/android/dialer/settings/GeneralSettingsFragment.java b/src/com/android/dialer/settings/GeneralSettingsFragment.java
index 63c6e33..4e04726 100644
--- a/src/com/android/dialer/settings/GeneralSettingsFragment.java
+++ b/src/com/android/dialer/settings/GeneralSettingsFragment.java
@@ -22,11 +22,11 @@ import android.os.Bundle;
 import android.os.Handler;
 import android.os.Message;
 import android.os.Vibrator;
-import android.preference.CheckBoxPreference;
 import android.preference.Preference;
 import android.preference.PreferenceCategory;
 import android.preference.PreferenceFragment;
 import android.preference.PreferenceScreen;
+import android.preference.SwitchPreference;
 import android.provider.Settings;
 
 import com.android.dialer.R;
@@ -55,8 +55,8 @@ public class GeneralSettingsFragment extends PreferenceFragment
     private Context mContext;
 
     private Preference mRingtonePreference;
-    private CheckBoxPreference mVibrateWhenRinging;
-    private CheckBoxPreference mPlayDtmfTone;
+    private SwitchPreference mVibrateWhenRinging;
+    private SwitchPreference mPlayDtmfTone;
     private Preference mRespondViaSms;
 
     private Runnable mRingtoneLookupRunnable;
@@ -80,8 +80,8 @@ public class GeneralSettingsFragment extends PreferenceFragment
         addPreferencesFromResource(R.xml.general_settings);
 
         mRingtonePreference = findPreference(BUTTON_RINGTONE_KEY);
-        mVibrateWhenRinging = (CheckBoxPreference) findPreference(BUTTON_VIBRATE_ON_RING);
-        mPlayDtmfTone = (CheckBoxPreference) findPreference(BUTTON_PLAY_DTMF_TONE);
+        mVibrateWhenRinging = (SwitchPreference) findPreference(BUTTON_VIBRATE_ON_RING);
+        mPlayDtmfTone = (SwitchPreference) findPreference(BUTTON_PLAY_DTMF_TONE);
         mRespondViaSms = findPreference(BUTTON_RESPOND_VIA_SMS_KEY);
 
         PreferenceCategory soundCategory = (PreferenceCategory) findPreference(CATEGORY_SOUNDS_KEY);
@@ -133,6 +133,9 @@ public class GeneralSettingsFragment extends PreferenceFragment
             boolean doVibrate = (Boolean) objValue;
             Settings.System.putInt(mContext.getContentResolver(),
                     Settings.System.VIBRATE_WHEN_RINGING, doVibrate ? 1 : 0);
+        } else if (preference == mPlayDtmfTone) {
+            Settings.System.putInt(mContext.getContentResolver(),
+                    Settings.System.DTMF_TONE_WHEN_DIALING, (Boolean) objValue ? 1 : 0);
         }
         return true;
     }
@@ -142,10 +145,7 @@ public class GeneralSettingsFragment extends PreferenceFragment
      */
     @Override
     public boolean onPreferenceTreeClick(PreferenceScreen preferenceScreen, Preference preference) {
-        if (preference == mPlayDtmfTone) {
-            Settings.System.putInt(mContext.getContentResolver(),
-                    Settings.System.DTMF_TONE_WHEN_DIALING, mPlayDtmfTone.isChecked() ? 1 : 0);
-        } else if (preference == mRespondViaSms) {
+        if (preference == mRespondViaSms) {
             // Needs to return false for the intent to launch.
             return false;
         }
-- 
2.5.0

